w3af_batch

```diff
@@ -0,0 +1,134 @@
+
+
+import logging
+import argparse
+import signal
+import threading as th
+import multiprocessing as mp
+import multiprocessing.dummy as mpd
+
+
+STOPSCAN = mp.Event()
+logger = logging.getLogger('w3af_batch')
+
+
+def cancel_scan(*args):
+    """
+    Dummy SIGINT hanlder which stops target list processing.
+    """
+    STOPSCAN.set()
+    raise KeyboardInterrupt('Scan is terminated by user request')
+
+
+def execute_scan(target, profile, timeout):
+    """
+    Execute w4af scan.  Try to stop it after timeout seconds.
+    """
+    def stop_scan(scanner):
+        """
+        Stop w3af scan after timeout.
+        """
+        if scanner.status.is_running():
+            scanner.stop()
+            logger.info('Scan of target %s is stopped', target)
+
+    from w3af.core.controllers.w3afCore import w3afCore
+    from w3af.core.data.kb.knowledge_base import kb
+
+    scanner = w3afCore()
+    scanner.profiles.use_profile(profile)
+    option_list = scanner.target.get_options()
+    option_list['target'].set_value(target)
+    scanner.target.set_options(option_list)
+    scanner.plugins.init_plugins()
+
+    timer = th.Timer(timeout, stop_scan, (scanner,))
+    timer.start()
+    scanner.start()
+    if timer.is_alive():
+        timer.cancel()
+        timer.join()
+        logger.debug('Target %s has %s vulnerabilities',
+                     target, len(kb.get_all_vulns()))
+
+
+def start_executor(target, profile, timeout):
+    """
+    Start w3af scan in a separate process. Terminate child process after
+    2 * timeout seconds.
+
+    :param target: Target uri
+    :param profile: Profile's internal name of path to pw3af file
+    :param timeout: Scan timeout
+    """
+    def stop_executor(executor, target):
+        """
+        Stop w3af scanner process after timeout.
+        """
+        if executor.is_alive():
+            executor.terminate()
+            logger.info('Scan of target %s is terminated', target)
+
+    executor = mp.Process(target=execute_scan,
+                          name='Executor',
+                          args=(target, profile, timeout))
+    timer = th.Timer(2 * timeout, stop_executor, (executor, target))
+    timer.start()
+    executor.start()
+    executor.join()
+    if timer.is_alive():
+        timer.cancel()
+        timer.join()
+
+
+def main(args):
+    """
+    Execute w3af scans of args.targets in parallel.  A pool of manager
+    threads is created. Each thread receives a target to scan. Manager
+    thread executes scan in a separate process. Scanner process will
+    try to stop w3af after timeout seconds. In addition manager thread
+    will terminate the child process if its execution time exceeds
+    2 x timeout seconds.
+    """
+
+    def terminate_scan(pool):
+        """
+        Terminate pool tasks processing in case of SIGINT.
+        """
+        if STOPSCAN.wait():
+            pool.terminate()
+
+    logger.info('Begin scanning')
+    pool = mpd.Pool(args.workers)
```

Comment:
Any special reason for this being a threadpool instead of a process pool? If you use a process pool you might be able to simplify your code (remove the mp.Process above)